### PR TITLE
Adjust n_conn to CONCURRENT_REQUESTS

### PR DESCRIPTION
**Pull request by Gallaecio**
Fixes https://github.com/scrapy-plugins/scrapy-zyte-api/issues/23

I thought about warning if a custom `client` is passed with an `n_conn` different than the one in `CONCURRENT_REQUESTS`, but since that could be done on purpose and providing a setting to silence that warning seemed overkill, I decided against it.

To do:

- [x] Figure out the issue with the 2 failing tests.

### CODE REVIEW

**Review by kmike — APPROVED** [review]
The code looks good, but I haven't run it. 

+1 to merge after confirming that the change allows to send more than 15 requests in parallel to Zyte API.

**Review by BurnzZ — APPROVED** [review]
LGTM @Gallaecio !

**Comment by BurnzZ** [comment]
Hi @Gallaecio ! Those are some productive improvements! 🎉 I think we can break them down into other PRs as they're out of scope for this PR. Aside from including this commit, +1 on merging this PR to master.

File: scrapy_zyte_api/handler.py [hunk 1]
```diff
@@ -25,28 +26,30 @@ def __init__(
...
+                    n_conn=settings.getint('CONCURRENT_REQUESTS'),
+                )
+            except NoApiKey:
+                logger.warning(
+                    "'ZYTE_API_KEY' must be set in the spider settings or env var "
+                    "in order for ScrapyZyteAPIDownloadHandler to work."
+                )
+                raise NotConfigured
+        self._client: AsyncClient = client
+        logger.info(f"Using Zyte API Key: {self._client.api_key[:7]}")
```
Comment (kmike) [inline scrapy_zyte_api/handler.py:43]: Would it make sense to include something like ..., to indicate that's not the full key?

**Comment by kmike** [comment]
Looks awesome @Gallaecio!

For me the main question is ZYTE_API_KEY handling. Let's say a spider sets a certain ZYTE_API_KEY in settings.py. User wants to use env variable. Is it right that previously it was possible by setting the ZYTE_API_KEY to "", and now it requires setting it to None, which is not possible from the command line?

**Review by kmike — APPROVED** [review]

**Comment by BurnzZ** [comment]
@kmike , commit https://github.com/scrapy-plugins/scrapy-zyte-api/commit/a255754c1cbd11cdf1cf026eeac026592f7dbb00 seems to be self-documenting enough. LGTM!

**Comment by Gallaecio** [comment]
> Let's say a spider sets a certain ZYTE_API_KEY in settings.py. User wants to use env variable. Is it right that previously it was possible by setting the ZYTE_API_KEY to "", and now it requires setting it to None, which is not possible from the command line?

Do we want to allow API keys to be empty strings?

- If we do, setting the `ZYTE_API_KEY` setting to an empty string should make `""` the API key. Maybe we can support a special string in settings that would translate to the key not being set, e.g. `NONE` or `UNSET`.
- If we do not, then this may be worth handling [upstream](https://github.com/zytedata/python-zyte-api/blob/main/zyte_api/apikey.py#L12-L20).

**Comment by kmike** [comment]
> Do we want to allow API keys to be empty strings?

Ideally, yes. But I think that's fine if scrapy-zyte-api doesn't allow them.

> If we do, setting the ZYTE_API_KEY setting to an empty string should make "" the API key. Maybe we can support a special string in settings that would translate to the key not being set, e.g. NONE or UNSET.

I'm not sure how to do it for scrapy-zyte-api, if only a string value can be set from cmdline (is it?). In this case, "UNSET" string doesn't look any better than "". If we're not talking about command line, None already works.

EIther way, it looks like a very minor edge case; I'm not against merging any of the implementations.

Testing failure looks like a large issue though, as it could mean tests become unstable; could you please check it?

**Comment by kmike** [comment]
Nice, thanks @Gallaecio!
